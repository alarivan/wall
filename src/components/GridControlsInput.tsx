import React from 'react';

interface Props {
  label: string;
  min: number;
  max: number;
  name: string;
  value: number;
  onChange: (value: number) => void;
}

const GridControlsInput: React.FC<Props> = ({
  label,
  min,
  max,
  name,
  value,
  onChange,
}) => {
  function handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
    e.preventDefault();
    const value: number = parseInt(e.target.value);
    if (value >= min && value <= max) {
      onChange(value);
    }
  }

  return (
    <>
      <label>{label}</label>
      <input
        data-testid='grid-controls-input'
        min={min}
        max={max}
        type='number'
        name={name}
        value={value}
        onChange={handleChange}
      />
    </>
  );
};

export default GridControlsInput;
