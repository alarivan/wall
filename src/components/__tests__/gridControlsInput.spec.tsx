import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import GridControlsInput from '../GridControlsInput';

afterEach(cleanup);

const props = {
  label: 'label',
  min: 5,
  max: 10,
  value: 8,
  name: 'name',
  onChange: jest.fn(),
};

describe('GridControlsInput', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<GridControlsInput {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onChange when value is in allowed range', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <GridControlsInput {...props} onChange={onChange} />,
    );

    fireEvent.change(getByTestId('grid-controls-input'), {
      target: { value: 9 },
    });

    fireEvent.change(getByTestId('grid-controls-input'), {
      target: { value: 20 },
    });

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
