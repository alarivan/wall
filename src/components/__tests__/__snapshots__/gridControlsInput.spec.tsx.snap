// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridControlsInput renders correctly 1`] = `
<DocumentFragment>
  <label>
    label
  </label>
  <input
    data-testid="grid-controls-input"
    max="10"
    min="5"
    name="name"
    type="number"
    value="8"
  />
</DocumentFragment>
`;
